framework/src/modules/nft/endpoint.ts

```diff
@@ -185,7 +197,11 @@ export class NFTEndpoint extends BaseEndpoint {
 			return { collectionExists: false };
 		}
 
-		const collectionID = Buffer.from(params.collectionID, 'hex');
+		const isNFTSupported = await this.isNFTSupported(context);
```

Comment:
Params in the context, as per the validation against schema definition `isCollectionIDSupportedRequestSchema` has only `chainID` and `collectionID`; however this call to `isNFTSupported` requires `id` field as per the schema definition `isNFTSupportedResponseSchema`; perhaps the call to `isNFTSupported` is not necessary here 🤔

Reply:
The context params contains the `id` (see below)

`params: {
          chainID: '27f38593',
          id: '34c3be55bcfcb8415fc60b0c3362a9a8',
          collectionID: '2cb34db9'
        },`
        
This `id` is being validated in the `isNFTSupported` method using the `isNFTSupportedRequestSchema.` as the context is passed as an arg to the method. 

The call to `isNFTSupported` in my opinion is important to correctly check if an NFT is supported as it takes into account the existence of the ALL_SUPPORTED_NFTS_KEY key in the SupportedNFTsStore.

Reply:
Hmm, this would suggest that the `id` field should be defined in the schema for the `isCollectionIDSupported` endpoint's request definition under `isCollectionIDSupportedRequestSchema`; now it defines only `chainID` and `collectionID`, as another endpoint `isSupportedNFT` is being called within this (`isCollectionIDSupported`) endpoint, shouldn't the parent endpoint's schema definition be a super set? However; even with this change, there is a chance of providing a `collectionID` that is different to the `collectionID`contained within the NFT ID provided through`id` field.

Reply:
If the context is the same that is passed to both methods, what are the chances that... `chance of providing a collectionID that is different to the collectionIDcontained within the NFT ID provided throughid field.` 🤔?

Reply:
I meant; since they are request parameters, they are allowed to be conflicting unless, the collectionID is sliced out of nftID and verified against the collectionID.